test/e2e/infinispan/encryption_test.go

```diff
@@ -45,6 +46,29 @@ func TestTLSWithExistingKeystore(t *testing.T) {
 	checkRestConnection(client_)
 }
 
+func TestTLSConditionWithBadKeystore(t *testing.T) {
+	t.Parallel()
+	defer testKube.CleanNamespaceAndLogOnPanic(t, tutils.Namespace)
+
+	// Create a resource without passing any config
+	spec := tutils.DefaultSpec(t, testKube, func(i *ispnv1.Infinispan) {
+		i.Spec.Security = ispnv1.InfinispanSecurity{
+			EndpointEncryption: tutils.EndpointEncryption(i.Name),
+		}
+	})
+
+	// Create secret
+	serverName := tutils.GetServerName(spec)
+	keystore, _ := tutils.CreateKeystore(serverName)
+	secret := tutils.EncryptionBadSecretKeystore(spec.Name, tutils.Namespace, keystore)
```

Comment:
Do you think we'll need this in another test? If not I would just declare inline to keep things simpler.

Reply:
there are private func inside. I changed the code to corrupt the right secret